### ISSUE TITLE
Autotracked sync relationship reference ids

### ISSUE THREAD

**Issue by Techn1x** [emberjs/data#7712]
### Reproduction

I did attempt to create a sandbox/twiddle for this, but it kind of relies on saving data and having an API, so it got too hard :(

### Description

Assuming a model has `@hasMany('item', { async: false }) items`

I seem to be unable to get a public way to access the `ids` of those items. My API always returns a `relationships` key with all the item ids - most of the time I do not want to load the full item records, but I often need to check if an item's id is in that list.

Calling `myRecord.items` returns the "your data is not loaded" error (expected, since this is marked as sync).

The way to do it in the past has been via `myRecord.hasMany('items').ids()`, but this does not seem to autotrack properly, when those ids are updated from a newer payload

Additionally it would be useful to be able to add/remove items from that and save the model
`record.items.pushObjects(items) && record.save()` would currently result in an error

Is there a way I can achieve this? The workaround I have at the moment is essentially
```js
get itemIds() {
  return this._internalModel.getManyArray('items').map((item) => item.id)
}
```

I have posted on discord about this, if you want further info
https://discord.com/channels/480462759797063690/486549196837486592/893294722599251989

### Versions

```cli
└─ ember-source@3.24.4
Done in 0.77s.
yarn list v1.22.10
warning Filtering by arguments is deprecated. Please use the pattern option instead.
└─ ember-cli@3.24.0
Done in 0.78s.
yarn list v1.22.10
├─ @ember-data/adapter@3.24.2
├─ @ember-data/canary-features@3.24.2
├─ @ember-data/debug@3.24.2
├─ @ember-data/model@3.24.2
├─ @ember-data/private-build-infra@3.24.2
├─ @ember-data/record-data@3.24.2
├─ @ember-data/rfc395-data@0.0.4
├─ @ember-data/serializer@3.24.2
├─ @ember-data/store@3.24.2
├─ babel-plugin-ember-data-packages-polyfill@0.1.2
└─ ember-data@3.24.2
```

**Comment by runspired** [comment]
closing as duplicate of #7102, we would like to get this addressed for 3.28+

**Comment by runspired** [comment]
#7796 will add this feature
